src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1934,6 +1934,10 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
     {
         auto table_function_ast = create.as_table_function->ptr();
         auto table_function = TableFunctionFactory::instance().get(table_function_ast, getContext());
+
+        if (!table_function->canBeUsedToCreateTable())
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Cannot create a table with *Cluster engine");
```

Comment:
```suggestion
            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Cannot create a table with {} engine, it can be used only as table function", table_function->getName());
```